### PR TITLE
global: new directory structure

### LINKED ISSUE #16: global: new directory structure
Restructure this REANA demo example to better separate input data, environment specification, analysis code, workflow definition, and other miscellanea such as documentation.

Use separate directories such as:

- code/foo.py
- data/input.dat
- documentation/notes.txt
- documentation/screenshot.png
- environment/Dockerfile
- workflow/workflow.cwl

We can also advertise the usage of separate directories for results and log files:

- results/myresult.cvs
- results/myplot.cvs
- logs/stdout.txt
- logs/stderr.txt
- logs/profile.txt

### PR DESCRIPTION
**Pull request by tiborsimko**
* Refactors directory structure of the hello world example to separate more
  clearly the code, the environment, the inputs, the outputs. (closes #16)

* Adds an input file with the names of the persons to greet to simulate input
  data file, making the example a little bit more realistic.

Signed-off-by: Tibor Simko <tibor.simko@cern.ch>

### CODE REVIEW

**Comment by tiborsimko** [comment]
@diegodelemos @hjhsalo This is WIP for illustration only.  Needs workflow amendments, then client example, and some verbiage updates...

File: README.rst [hunk 1]
```diff
@@ -15,125 +15,137 @@ About
...
-and test whether it works locally:
+For example:
 
 .. code-block:: console
 
-    $ docker run -v `pwd`:/code -i -t --rm helloworld python helloworld.py --name Jim --sleeptime 1
-    $ tail -1 output/greetings.txt
-    Hello Jim!
+    $ cat environment/Dockerfile
+    FROM python:2.7
```
Comment (diegodelemos) [inline README.rst]: As an example of a dependency in an additional Python package we could use a library that makes the `helloworld.py` as small as possible, such as `click` so we can profit of removing boilerplate code.. but not for now.

File: README.rst [hunk 2]
```diff
@@ -15,125 +15,137 @@ About
...
+        - code/helloworld.py
+    inputs:
+      - files:
+        - inputs/names.txt
+      - parameters:
+        - sleeptime: 2
+    outputs:
+      - files:
+        - outputs/greetings.txt
+    environment:
```
Comment (tiborsimko) [inline README.rst]: @diegodelemos @hjhsalo Shall we go aleady for the plural option?

```yaml
environments:
  - name: python
    type: docker
    image: python:2.7
  - name: fitting
    type: docker
    image: johndoe/some-compiled-environment
  - name: plotting
    type: docker
    image: reanahub/reana-env-root6
```

(`name` is optional, could probably leave it out... should be enough)

Comment (hjhsalo) [reply README.rst]: Sure, but just as a disclaimer, I don't know enough about yadage to say how much work to support this needs or if this works out-of-the-box

Comment (diegodelemos) [reply README.rst]: Yes, let’s do it now

Comment (tiborsimko) [reply README.rst]: @hjhsalo Yadage won't see this, and the environments will be written in its yaml file directly anyway.  But `reana-client` might need some amendments later...

**Comment by tiborsimko** [comment]
@diegodelemos Merged, please prepare a separate PR with yadage workflow spec and the `reana-client` usage scenario on top of this.

**Pull request merged**